### DISCUSSION TITLE
QSelect display values are not rendered

### DISCUSSION THREAD

**Discussion by wimvanhouts** [quasarframework/quasar#15370] [General - Components / Directives / etc] [answered]
I have spent last day search to get a QSelect component working with HTML rendering of the options and the display value. No matter what I do, the display value never renders the HTML and displays the sanitized content. 

As far as I understand the documentation, if the options are HTML and rendered correctly, than the value should be displayed as well. See also docu on sanitizing https://quasar.dev/vue-components/select#sanitization which states that the displayed value is displayed as html if options-html property of the QSelect is set. This is not the case in my page, and I really cannot see why. I have tried any combination of option and display attributes... none of them work. I search for a bug as well in the issues list but couldn't find one in that  direction. 

My code is currently the below one:

```
<!-- Notary selection dropdown -->
  <q-select
          class="q-mt-md"
          filled
          use-input
          v-model="viewState.selectedNotary"
          hide-selected
          fill-input
          input-debounce="0"
          :options="viewState.availableNotaries"
          options-html
          option-label="htmlcard"
          @filter="filterNotaries"
          @filter-abort="abortFilterNotaries"
          :hint="$t('view.notary.selectnotary')"
          hide-hint
          style="width: 500px; height:150px padding-bottom: 32px"
        >
          <template v-slot:no-option>
            <q-item>
              <q-item-section class="text-grey">
                {{ $t('view.notary.noresults') }}
              </q-item-section>
            </q-item>
          </template>
        </q-select>
```
This works for the options, they are shown as:

![image](https://user-images.githubusercontent.com/12234274/216362357-de797361-5240-422e-b0b1-f595e45f5fbd.png)

However, when I then select an item in this list, it looks as below

![image](https://user-images.githubusercontent.com/12234274/216362856-fb7bd30b-fc68-4b47-85e4-7aafcf844c95.png)

Why doesn't it render as HTML?

**Comment by pdanpdan** [comment]
display-value-html, written in the same place in docs

**Reply by wimvanhouts** [reply]
Yes, read that too, and tried it, but, it doesn't change the outcome. Same display as above images when I used the following code

```
       <q-select
          class="q-mt-md"
          filled
          use-input
          v-model="viewState.selectedNotary"
          hide-selected
          fill-input
          input-debounce="0"
          :options="viewState.availableNotaries"
          options-html
          option-label="htmlcard"
          display-value-html
          @filter="filterNotaries"
          @filter-abort="abortFilterNotaries"
          :hint="$t('view.notary.selectnotary')"
          hide-hint
          style="width: 500px; height:150px padding-bottom: 32px"
        >
          <template v-slot:no-option>
            <q-item>
              <q-item-section class="text-grey">
                {{ $t('view.notary.noresults') }}
              </q-item-section>
            </q-item>
          </template>
        </q-select>
```

**Comment by wimvanhouts** [comment]
tried with display value and display value html, and even then no changes

![image](https://user-images.githubusercontent.com/12234274/216446331-45f419da-8b7f-40b1-b5b3-77f791928ba3.png)

Strange thing: I added just like in the documentation Company: in front of the display-value property, just for testing purposes and I see in the dev tools the value is set, but I don't see the Company prefix rendered in the browser. Thus it seems as if the display value is not used?

**Comment by pdanpdan** [comment]
Please make a codepen

**Comment by wimvanhouts** [comment, marked as answer]
Found the cause in the meantime. Combination of using input to get a filtered options list and HTML displaying seems not be working as I expected. Tried to do this in one QSelect but this seems not to be possible... there is also not an example of this, probably with good reasons. In retrospect, that might be "logical" since you need somewhere to input text, but, also need to display the HTML. The hide-selected option here is needed to show the input fieId to type, but, if you remove it you get half of the control showing the HTML as expected, and half of the control showing an input field... which is correct but really strange in my scenario. Will rethink my UI design and solve this differently. Thx for replying

**Comment by pdanpdan** [comment]
Yes, if you want html with `use-input hide-selected fill-input` it's a little bit difficult
You need to:
- use objects for values
- optional - if you want to keep only the value use `map-options emit-value` - else extract your value from model when you need it
- for value key use something you need
- for label use the text you want in input (editable selection)
- add another key with the html
- use an `option slot` to display with `v-html` what is inside your html key
- optional - if you don't use `hide-selected` also use a `selected-item slot` to display the html from selected items
